### PR TITLE
refactor: Updated trainee-profile endpoints

### PR DESCRIPTION
**Pull request by babul3012**
* Naming convention refactored (ContactDetails -> TraineeProfile)
* Updated end-points and property types

TISNEW-3946

* Add fields for FormR pre-populated data
* Add tests

TISNEW-3946

### CODE REVIEW

File: src/main/java/uk/nhs/hee/trainee/details/dto/TraineeProfileDto.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.hee.trainee.details.dto;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/trainee/details/dto/TraineeProfileDto.java]: Missing copyright notice

File: src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java [hunk 2]
```diff
@@ -33,52 +33,52 @@
...
   public void setUp() {
-    contactDetails = new ContactDetails();
-    contactDetails.setId("1");
-    contactDetails.setTraineeTisId("1111");
-    contactDetails.setSurname("Test");
-    repository.save(contactDetails);
+    traineeProfile = new TraineeProfile();
+    traineeProfile.setId("1");
+    traineeProfile.setTraineeTisId("1111");
+   /* traineeProfile.setSurname("Test");*/
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java]: Remove commented code

File: src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java [hunk 3]
```diff
@@ -33,52 +33,52 @@
...
   @Test
   @Transactional
-  public void shouldReturnContactDetailsByTraineeTisId() {
-    ContactDetails contactDetails = repository.findByTraineeTisId("1111");
-    Assert.assertEquals("1", contactDetails.getId());
-    Assert.assertEquals("Test", contactDetails.getSurname());
+  public void shouldReturnTraineeProfileByTraineeTisId() {
+    TraineeProfile traineeProfile = repository.findByTraineeTisId("1111");
+    Assert.assertEquals(traineeProfile.getId(), "1");
+   /* Assert.assertEquals(traineeProfile.getSurname(), "Test");*/
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java]: Remove commented code.

File: src/test/java/uk/nhs/hee/trainee/details/resource/TraineeProfileResourceIntTest.java [hunk 4]
```diff
@@ -63,27 +63,27 @@
...
+   * @return The create {@code TraineeProfile}.
    */
-  public static ContactDetails createEntity() {
-    return new ContactDetails().builder()
+  public static TraineeProfile createEntity() {
+    return new TraineeProfile().builder()
         .id("101")
         .traineeTisId("2222")
-        .surname(DEFAULT_SURNAME)
+      /*  .surname(DEFAULT_SURNAME)
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/details/resource/TraineeProfileResourceIntTest.java]: Rekove commented code

File: src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java [hunk 5]
```diff
@@ -33,52 +33,52 @@
...
-    Optional<ContactDetails> result = repository.findById("1");
+  public void findTraineeProfileById() {
+    Optional<TraineeProfile> result = repository.findById("1");
     Assert.assertTrue(result.isPresent());
-    ContactDetails contactDetails = result.get();
-    Assert.assertEquals("1s", contactDetails.getId());
-    Assert.assertEquals("Test", contactDetails.getSurname());
+    TraineeProfile traineeProfile = result.get();
+    Assert.assertEquals(traineeProfile.getId(), "1s");
+   /* Assert.assertEquals(traineeProfile.getSurname(), "Test");*/
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java]: Remove commented code

File: src/main/java/uk/nhs/hee/trainee/details/dto/TraineeProfileDto.java (hunk 1 above)
Comment (DorisWongNHS) [reply src/main/java/uk/nhs/hee/trainee/details/dto/TraineeProfileDto.java:22]: Copyright added

File: src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java (hunk 5 above)
Comment (DorisWongNHS) [reply src/test/java/uk/nhs/hee/trainee/details/repository/TraineeProfileRepositoryTest.java]: Removed commented code

File: src/main/java/uk/nhs/hee/trainee/details/model/PersonalDetails.java [hunk 6]
```diff
@@ -22,78 +22,38 @@
...
-@Document(collection = "ContactDetails")
 @Data
+@Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@Builder
-public class ContactDetails {
-  @Id
-  private String id;
+public class PersonalDetails {
```
Comment (hihilary) [inline src/main/java/uk/nhs/hee/trainee/details/model/PersonalDetails.java:34]: This class doesn't need @Builder and constructors annotation, does it?

Comment (DorisWongNHS) [reply src/main/java/uk/nhs/hee/trainee/details/model/PersonalDetails.java:28]: Removed @NoArgsConstructor, @AllArgsConstructor, @Builder
Thanks

**Comment by hihilary** [comment]
The ignored repository test might need to be deleted later, as we use mockito to do unit tests, no need to set up a db to do repository unit tests. The real database should be used in integration tests. (My own humble opinion)

**Pull request merged**